packages/teams-js/src/public/visualMedia.ts

```diff
@@ -0,0 +1,274 @@
+   *
+   * @beta
+   * @returns Promise that will resolve with true if the user had granted the app permission to media information, or with false otherwise,
+   * In case of an error, promise will reject with the error. Function can also throw a NOT_SUPPORTED_ON_PLATFORM error
+   */
+  export function hasPermission(): Promise<boolean> {
+    ensureInitialized(runtime, FrameContexts.content, FrameContexts.task);
+    if (!isSupported()) {
+      throw errorNotSupportedOnPlatform;
+    }
+    const permissions: DevicePermission = DevicePermission.Media;
+
+    return new Promise<boolean>((resolve) => {
+      resolve(sendAndHandleSdkError('permissions.has', permissions));
+    });
+  }
+
+  /**
+   * Requests user permission for visualMedia
+   * @beta
+   * @returns Promise that will resolve with true if the user consented permission for media, or with false otherwise,
+   * In case of an error, promise will reject with the error. Function can also throw a NOT_SUPPORTED_ON_PLATFORM error
+   */
+  export function requestPermission(): Promise<boolean> {
+    ensureInitialized(runtime, FrameContexts.content, FrameContexts.task);
+    if (!isSupported()) {
+      throw errorNotSupportedOnPlatform;
+    }
+    const permissions: DevicePermission = DevicePermission.Media;
+
+    return new Promise<boolean>((resolve) => {
+      resolve(sendAndHandleSdkError('permissions.request', permissions));
+    });
+  }
+
+  /**
+   * Checks if visualMedia capability is supported by the host
+   * @returns boolean to represent whether media is supported
+   * @beta
+   * @throws Error if {@linkcode app.initialize} has not successfully completed
+   */
+  export function isSupported(): boolean {
+    return ensureInitialized(runtime) && runtime.supports.visualMedia && runtime.supports.permissions ? true : false;
+  }
+
+  /**
+   * To enable this image capability will let the app developer ask the user to get images from camera/local storage
+   * @beta
+   */
+  export namespace image {
+    /**
+     * Capture one or multiple image(s) throughing camera.
+     * @beta
+     * @param imageInputs - The input params to customize the image(s) to be captured
+     * @returns Promise that will resolve with {@link VisualMediaFile[]} object or reject with an error.
+     */
+    export function captureImages(imageInputs: ImageProperties): Promise<VisualMediaFile[]> {
+      ensureInitialized(runtime, FrameContexts.content, FrameContexts.task);
+      if (!isSupported()) {
+        throw errorNotSupportedOnPlatform;
+      }
+      if (!imageInputs || imageInputs == null || imageInputs.visualMediaCount > 10) {
+        throw new Error('Must supply the valid image(s)');
+      }
+      // waiting and return the response from hub-SDK
+      return sendAndHandleSdkError<VisualMediaFile[]>('visualMedia.image.captureImages', imageInputs);
+    }
+
+    /**
+     * Upload the existing image(s) from camera roll to the mos app.
```

Comment:
>mos app. [](http://example.com/codeflow?start=60&length=8)

"mos app" is not a phrase that developers know

Reply:
removed that part, thanks for pointing that out!